### PR TITLE
Add extensibility documentation

### PR DESCRIPTION
**Pull request by xiaoheng1**
Add extensibility documentation

### CODE REVIEW

File: content/zh/docs/v3.0/concepts/extensibility.md [hunk 1]
```diff
@@ -4,4 +4,59 @@ title: "如何扩展 Dubbo"
 linkTitle: "扩展性"
 weight: 6
 description: "Dubbo 通过 SPI 机制提供了非常灵活的可扩展性"
----
\ No newline at end of file
+---
+
+## 扩展设计理念
+
+可扩展性是任何一个系统所追求的，对于 Dubbo 来说是同样适用。
```
Comment (pinxiong) [inline content/zh/docs/v3.0/concepts/extensibility.md:11]: I think this part needs to be strengthed, as follows:

+ What's the system extension?
+ What's the benifits of system extension?
+ What are the commonly used extension mechanisms?

File: content/zh/docs/v3.0/concepts/extensibility.md [hunk 2]
```diff
@@ -4,4 +4,59 @@ title: "如何扩展 Dubbo"
...
+
+### 微核心，插件式，平等对待第三方
```
Comment (pinxiong) [inline content/zh/docs/v3.0/concepts/extensibility.md:13]: This part should belong to the above chapter `What are the commonly used extension mechanisms?`

File: content/zh/docs/v3.0/concepts/extensibility.md [hunk 3]
```diff
@@ -4,4 +4,59 @@ title: "如何扩展 Dubbo"
...
+
+一般来说，系统会采用 Factory、IoC、OSGI 等方式管理插件生命周期。考虑到 Dubbo 的适用面，不想强依赖 Spring 等 IoC 容器。
+而自己造一个小的 IoC 容器，也觉得有点过度设计，所以选择最简单的 Factory 方式管理插件。在 Dubbo 中，所有内部实现和第三方实现都是平等的。
+
+### 每个扩展点只封装一个变化因子，最大化复用
+
+每个扩展点的实现者，往往都只是关心一件事。如果用户有需求需要进行扩展，那么只需要对其关注的扩展点进行扩展就好，极大的减少用户的工作量。
```
Comment (pinxiong) [inline content/zh/docs/v3.0/concepts/extensibility.md:20]: Maybe, this part should belong to the above chapter `What's the benifits of system extension?`

File: content/zh/docs/v3.0/concepts/extensibility.md [hunk 4]
```diff
@@ -4,4 +4,59 @@ title: "如何扩展 Dubbo"
...
+
+## 扩展设计目标
+
```
Comment (pinxiong) [inline content/zh/docs/v3.0/concepts/extensibility.md:23]: This part should focus on Dubbo's feature more

+ What are the capabilities of Dubbo's extension? 
+ How to design?
+ How to work? We'd better give a detailed example

File: content/zh/docs/v3.0/concepts/extensibility.md (hunk 1 above)
Comment (xiaoheng1) [reply content/zh/docs/v3.0/concepts/extensibility.md:11]: done

File: content/zh/docs/v3.0/concepts/extensibility.md (hunk 2 above)
Comment (xiaoheng1) [reply content/zh/docs/v3.0/concepts/extensibility.md]: done

File: content/zh/docs/v3.0/concepts/extensibility.md (hunk 3 above)
Comment (xiaoheng1) [reply content/zh/docs/v3.0/concepts/extensibility.md]: done

File: content/zh/docs/v3.0/concepts/extensibility.md (hunk 4 above)
Comment (xiaoheng1) [reply content/zh/docs/v3.0/concepts/extensibility.md:22]: done

**Pull request merged**